### ISSUE TITLE
aws_service_discovery_public_dns_namespace name length limit

### ISSUE THREAD

**Issue by ddriddle** [terraform-providers/terraform-provider-aws#5339]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.7
+ provider.aws v1.28.0

### Affected Resource(s)

* aws_service_discovery_public_dns_namespace

### Terraform Configuration Files

```hcl
provider "aws" {
  region  = "us-east-2"
  version = "1.28.0"
}

resource "aws_service_discovery_public_dns_namespace" "default" {
  name = "very.very.long.domain.name.example.com"
}
```

### Debug Output

https://gist.github.com/ddriddle/5ec7170b26020657fc2a2ee69ca41e54

### Expected Behavior

Should have created the resource.

### Actual Behavior

```
* aws_service_discovery_public_dns_namespace.default: 1 error(s) occurred:

* aws_service_discovery_public_dns_namespace.default: InvalidInput: 1 validation error detected: Value 'tf-very.very.long.domain.name.example.com20180725230632994500000001' at 'creatorRequestId' failed to satisfy constraint: Member must have length less than or equal to 64
	status code: 400, request id: 5f661afa-905f-11e8-9197-5bd16439433f
```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

Appears to be related to closed issue #4701.

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.33.0 of the AWS provider, later this week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.33.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1330-august-22-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
